### PR TITLE
Generate API docs with codox

### PR DESCRIPTION
**Pull request by Jannis**

### CODE REVIEW

File: build.boot [hunk 1]
```diff
@@ -91,16 +86,20 @@
...
   []
   (merge-env! :source-paths #{"src/test"})
   identity)
 
+(deftask docs
+  []
+  (comp
+   (codox :name "workflo/macros"
+          :source-paths #{"src/main"}
+          :output-path "api-docs"
```
Comment (yanivtal) [inline build.boot:99]: How about just docs?

Comment (Jannis) [reply build.boot:99]: Could do, the problem is that's where our more tutorial-like Markdown documentation for the different macros resides. We could move that somewhere else of course.

**Comment by yanivtal** [comment]
Should those just be combined into the main readme?

**Comment by Jannis** [comment]
I think that would make the main README too long to be readable. GitHub wiki pages are another alternative...

**Pull request merged**